File: Python/172.Factorial Trailing Zeroes.py
#求n！末尾有几个0
class Solution(object):
    def trailingZeroes(self, n):
        result = 0
        while n>0:
            n = n//5
            result += n
        return result

s = Solution()
print(s.trailingZeroes(124))

File: C++/Math/172.Factorial Trailing Zeroes.cpp
/*
求n!末尾0的个数。
*/

#include <iostream>
#include <cstring>
using namespace std;

class Solution {
public:
	int trailingZeroes(int n) {
        int result = 0;
        while(n)
        {
        	n /= 5;
        	result += n;
        }
        return result;
    }
};
int main()
{
	Solution s;
	cout<<s.trailingZeroes(130)<<endl;
}

